src/vscode-bicep/src/commands/deploy.ts

```diff
@@ -0,0 +1,317 @@
+      await this.treeManager.azManagementGroupTreeItem.showTreeItemPicker<LocationTreeItem>(
+        "",
+        context
+      );
+    const managementGroupId = managementGroupTreeItem.id;
+
+    if (managementGroupId) {
+      const location = await vscode.window.showInputBox({
+        placeHolder: "Please enter location",
+      });
+
+      if (location) {
+        const parameterFilePath = await this.selectParameterFile(
+          context,
+          documentUri
+        );
+
+        await this.sendDeployCommand(
+          textDocument,
+          parameterFilePath,
+          managementGroupId,
+          deploymentScope,
+          location,
+          template
+        );
+      }
+    }
+  }
+
+  private async handleResourceGroupDeployment(
+    context: IActionContext,
+    textDocument: TextDocumentIdentifier,
+    documentUri: vscode.Uri,
+    deploymentScope: string,
+    template: string
+  ) {
+    const resourceGroupTreeItem =
+      await this.treeManager.azResourceGroupTreeItem.showTreeItemPicker<AzResourceGroupTreeItem>(
+        "",
+        context
+      );
+    const resourceGroupId = resourceGroupTreeItem.id;
+
+    if (resourceGroupId) {
+      const parameterFilePath = await this.selectParameterFile(
+        context,
+        documentUri
+      );
+
+      await this.sendDeployCommand(
+        textDocument,
+        parameterFilePath,
+        resourceGroupId,
+        deploymentScope,
+        "",
+        template
+      );
+    }
+  }
+
+  private async handleSubscriptionDeployment(
+    context: IActionContext,
+    textDocument: TextDocumentIdentifier,
+    documentUri: vscode.Uri,
+    deploymentScope: string,
+    template: string
+  ) {
+    const locationTreeItem =
+      await this.treeManager.azLocationTree.showTreeItemPicker<LocationTreeItem>(
+        "",
+        context
+      );
+    const location = locationTreeItem.label;
+    const subscriptionId = locationTreeItem.subscription.subscriptionPath;
+    const parameterFilePath = await this.selectParameterFile(
+      context,
+      documentUri
+    );
+
+    await this.sendDeployCommand(
+      textDocument,
+      parameterFilePath,
+      subscriptionId,
+      deploymentScope,
+      location,
+      template
+    );
+  }
+
+  private async sendDeployCommand(
+    textDocument: TextDocumentIdentifier,
+    parameterFilePath: string,
+    id: string,
+    deploymentScope: string,
+    location: string,
+    template: string
+  ) {
+    const bicepDeployParams: BicepDeployParams = {
+      textDocument,
+      parameterFilePath,
+      id,
+      deploymentScope,
+      location,
+      template,
+    };
+    const deploymentResponse: string = await this.client.sendRequest(
+      bicepDeployRequestType,
+      bicepDeployParams
+    );
+
+    this.outputChannelManager.appendToOutputChannel(deploymentResponse);
+  }
+
+  private async selectParameterFile(
+    _context: IActionContext,
+    sourceUri: Uri | undefined
+  ): Promise<string> {
```

Comment:
It would be more normal to return Promise<string|undefined>.  If you can't pass undefined through the command layer to langserver, I'd prefer to see you change it to empty (paramPath ?? "") only at that point, and return undefined here for no param file.

Reply:
Updated here:
https://github.com/Azure/bicep/commit/c8c34f96479acc79ea4e6280aa65fb932b1ca3ab

Thanks!